math/iterative_factorial.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file
+ * @brief Iterative implementation of factorial
+ *
+ * @author [Renjian-buchai](https://github.com/Renjian-buchai)
+ *
+ * @description Calculates factorial iteratively.
+ *
+ * @example
+ * int main() {
+ *     using std::cout;
+ *
+ *     cout << "10! = " math::iterativeFactorial(10);
+ * }
+ *
+ * Output:
+ * 10! = 3,628,800
+ *
+ */
+
+#include <cassert>
+#include <exception>
+#include <iostream>
+
+/**
+ * @namespace
+ * @brief Mathematical algorithms
+ */
+namespace math {
+
+/**
+ * @brief Calculates the factorial iteratively.
+ * @param n Nth factorial.
+ * @return Factorial.
+ * @note 0! = 1.
+ * @warning Maximum=20 because there are no 128-bit integers in C++. 21!
+ * returns 1.419e+19, which is not 21! but (21! % UINT64_MAX).
+ */
+uint64_t iterativeFactorial(uint8_t n) {
+    if (n > 20) {
+        throw new std::invalid_argument("Maximum n value is 20");
+    }
+
+    // 1 because it is the identity number of multiplication.
```

Comment:
this might be a triple slash ***not sure***

Reply:
Based on what I observed, triple slashes are for documentation while double slashes are just explanations on why it's there. Since most other algorithms use it in this way, I _think_ double slashes should be used here.

Reply:
makes sense